### ISSUE TITLE
How to reshape image into vectors with channels?

### ISSUE THREAD

**Issue by pwodzu** [Microsoft/CNTK#576]
Hi,

I want to reshape the output of convolution with dimensions [100 x 100 x 32] into matrix that contains 32 rows with 10000 values in each of them. Can someone help me how to do this?

Regards,
Piotr

**Comment by frankseide** [comment]
CNTK is column-major, so this is not just a reshape, but also requires a transposition. The first step is to reshape to [10000 x 32]:

    y = FlattenDimensions (x, 1/*first axis*/, 2/*num axes*/)   # [100 x 100 x 32] --> [(100*100) x 32]

and then transpose it:

    z = Transpose (y)

FlattenDimensions() only exists in BrainScript, not in the older `NDLNetworkBuilder`. The old NDL has a `Reshape()` operation, but it seems it only supports rank-3 tensors (while you want to go from 3 to 2).

**Comment by pwodzu** [comment]
Thank you very much for the answer. I would like to ask if the BrainScript is supported in MEL?

**Comment by frankseide** [comment]
It works differently, and is still somewhat under construction hence no documentation. What would you like to do?

**Comment by pwodzu** [comment]
I would like to use the trained CN model in my CNN. In this case i plan to join this trained model with aim to be the simple "driver" for the convolutional layer/s in the new CNN structure. As the consequence I want to use the approach, that you suggested in earlier post. Thanks a lot!

Regards,
Piotr

**Comment by pwodzu** [comment]
I try to use the FlattenDimensions input [100 x 100 x 32 x *] as below
y = FlattenDimensions(inp, 10000, 32) 

Unfortunatelly, I get the exception:
[CALL STACK]
    > Microsoft::MSR::CNTK::ComputationNetwork::  ValidateNode
    - Microsoft::MSR::CNTK::ComputationNetwork::  ValidateNodes
    - Microsoft::MSR::CNTK::ComputationNetwork::  ValidateNetwork
    - Microsoft::MSR::CNTK::ComputationNetwork::  CompileNetwork
    - Microsoft::MSR::CNTK::ComputationNetwork::  ConstructFromRoots
    - Microsoft::MSR::CNTK::ComputationNetwork::  ComputationNetwork
    - std::make_shared<Microsoft::MSR::CNTK::ComputationNetwork,std::shared_ptr<Microsoft::MSR::ScriptableObjects::IConfigRecord> const & __ptr64>
    - Microsoft::MSR::ScriptableObjects::MakeRuntimeObject<Microsoft::MSR::CNTK::ComputationNetwork>
    - <lambda_56a46857eb9bf0001b5ffab01c6f03ed>::  operator  ()
    - std::_Callable_obj<<lambda_56a46857eb9bf0001b5ffab01c6f03ed>,0>::_ApplyX<std::shared_ptr<Microsoft::MSR::ScriptableObjects::Object>,std::shared_ptr<Microsoft::MSR::ScriptableObjects::IConfigRecord>>
    - std::_Func_impl<std::_Callable_obj<<lambda_56a46857eb9bf0001b5ffab01c6f03ed>,0>,std::allocator<std::_Func_class<std::shared_ptr<Microsoft::MSR::ScriptableObjects::Object>,std::shared_ptr<Microsoft::MSR::ScriptableObjects::IConfigRecord>>>,std::  shared_pt
    - std::_Func_class<std::shared_ptr<Microsoft::MSR::ScriptableObjects::Object>,std::shared_ptr<Microsoft::MSR::ScriptableObjects::IConfigRecord>>::  operator  ()
    - Microsoft::MSR::BS::  Evaluate
    - <lambda_a048d19f5114b6bccccaea8ea1203939>::  operator  ()
    - std::_Callable_obj<<lambda_a048d19f5114b6bccccaea8ea1203939>,0>::_ApplyX<Microsoft::MSR::ScriptableObjects::ConfigValuePtr>
    - std::_Func_impl<std::_Callable_obj<<lambda_a048d19f5114b6bccccaea8ea1203939>,0>,std::allocator<std::_Func_class<Microsoft::MSR::ScriptableObjects::ConfigValuePtr>>,Microsoft::MSR::ScriptableObjects::ConfigValuePtr>::  _Do_call

EXCEPTION occurred: SmallVector: index overflow

How to use FlattenDimensions properly?

**Comment by pinzhang10** [comment]
@frankseide 

Hi Frankseide

I am using SplitDimension(labels,1,2). But it says "EXCEPTION occurred: Undefined function or macro 'SplitDimension' in SplitDimension(labels,1,2)"

Should I include something to make it work? I tried to include CNTK.core.bs, but it didn't work. 

Thank you for your help!

**Comment by pinzhang10** [comment]
I can not use the function is CNTK.core.bs file. It says Undefined function or macro 'Splice' 

I cloned the CNTK code yesterday and compiled by myself.

Does anybody know why?

**Comment by frankseide** [comment]
I think you are using the old NDLNetworkBuilder. The error message from BrainScript would be different.

If you have an existing NDL file, please see [here](https://github.com/Microsoft/CNTK/wiki/BrainScript-Network-Builder#legacy-ndlnetworkbuilder) for how to convert it. With luck it will be easy. (We are still in the process of updating our examples.)

**Comment by pinzhang10** [comment]
@frankseide I updated from NDLNetworkBuilder to BrainScriptNetworkBuilder. 
But then, the function in Shared.bs I downloaded doesn't work.

It says

<<<<<<<<<<<<<<<<<<<< PROCESSED CONFIG WITH ALL VARIABLES RESOLVED <<<<<<<<<<<<<<<<<<<<
Commands: train output test
Precision = "float"
CNTKModelPath: ../Output/Models/06_DeConvPooling
CNTKCommandTrainInfo: train : 50
CNTKCommandTrainInfo: CNTKNoMoreCommands_Total : 50

##############################################################################
#                                                                            #
# Action "train"                                                             #
#                                                                            #
##############################################################################

CNTKCommandTrainBegin: train

Creating virgin network.

[CALL STACK]
      conv32 = ConvBNReLULayer(conv31, cMap3, size32, kW, kH, hStride, vStride, wScale, bValue, scValue, bnTimeConst)
                              a
  ].y
    9
      y = RectifiedLinear(c)
                         8
      Relu(_, tag='') = new ComputationNode [ operation = 'RectifiedLinear' ; inputs = _ /*plus the function args*/ ]
                        7                                                              6
      y = RectifiedLinear(c)
                          5
      c = ConvBNLayer(inp, outMap, inWCount, kW, kH, hStride, vStride, wScale, bValue, scValue, bnTimeConst)
                     4
  ].c
    3
      c = ConvBNLayerW(W, inp, outMap, kW, kH, hStride, vStride, bValue, scValue, bnTimeConst)
                      2
  ].y
    1
      y = BatchNormalization(c, sc, b, m, isd, eval=false, spatial=true, normalizationTimeConstant=bnTimeConst , imageLayout="$imageLayout$")
                            0^
EXCEPTION occurred while evaluating: shared.bs(67): function expects 6 positional parameters, 5 were provided

I did not change this function.  Thank you for your help.

**Comment by frankseide** [comment]
Could you let me know where Shared.bs comes from, and also 06_DeConvPooling?

**Comment by pinzhang10** [comment]
@frankseide 
Thank you for your quick reply!

the Shared.bs is downloaded from https://github.com/Microsoft/CNTK/blob/master/Examples/Image/MNIST/Config/Shared.bs

The 06_DeconvPooling an is as below (I am testing the new BrainScriptNetworkBuilder I made)

 imageW = 32
    imageH = 36
    imageC = 1
    labelDim = 1152
 
    kW = 5
    kH = 5
    hStride = 1
    vStride = 1
    size1 = 25
    wScale = 0.5
    bValue = 0

	scValue = 1
    bnTimeConst = 0

    features = ImageInput(imageW, imageH, imageC, imageLayout="$imageLayout$")
    labels = Input((imageW:imageH:imageC), tag="label") 
    featScale = Constant(0.0006)
    featScaled = Scale(featScale, features)


    # conv1
    # weight[cMap1, kW1 * kH1 * inputChannels]
    conv1 = ConvBNReLULayer(featScaled, cMap1, size1, kW, kH, hStride, vStride, wScale, bValue, scValue, bnTimeConst)
	
    mse = SquareError(labels , conv1)
    FeatureNodes = (features)
    CriterionNodes = (mse)


The cntk file is as below

# Parameters can be overwritten on the command line
# for example: cntk configFile=myConfigFile RootDir=../.. 
# For running from Visual Studio add
# currentDirectory=$(SolutionDir)/<path to corresponding data folder> 
RootDir = ".."

ConfigDir = "$RootDir$/Config"
DataDir   = "$RootDir$/Data"
OutputDir = "$RootDir$/Output"
ModelDir  = "$OutputDir$/Models"

#deviceId = 0
imageLayout = "cudnn"
# Override the above as follows when running on CPU:
 deviceId = 0

command = train

precision = "float"
modelPath = "$ModelDir$/06_DeConvPooling"

# uncomment the following line to write logs to a file 
stderr = "$OutputDir$/06_DeConvPooling_out"
traceLevel=1
numMBsToShowResult=1

prefetch=true

# If set to true, always initialize the network on CPU, making initialization consistent across CPU and GPU targets (for testing).
initOnCPUOnly=true

#######################################
#  TRAINING CONFIG                    #
#######################################

train = [
    action = "train"

    BrainScriptNetworkBuilder = (new ComputationNetwork [ 
	include "shared.bs"       
    include "$ConfigDir$/06_DeConvPooling.bs"
])
    
    SGD = [
        epochSize = 0
        minibatchSize = 17
        learningRatesPerMB = 0.0002
        momentumPerMB = 0.9
        maxEpochs = 50
    ]
    
    reader = [
        readerType = "CNTKTextFormatReader"
        file = "$DataDir$/Train-Nodule-32x36.txt"
        input = [
            features = [
                dim = 1152
                format = "dense"
            ]
            labels = [
                dim = 1152
                format = "dense"
            ]
        ]
    ]   
]


Thank you!

**Comment by pinzhang10** [comment]
@frankseide 
Thank you for your quick reply!

the Shared.bs is downloaded from https://github.com/Microsoft/CNTK/blob/master/Examples/Image/MNIST/Config/Shared.bs

The 06_DeconvPooling.bs is as below (I am testing the new BrainScriptNetworkBuilder I made)

imageW = 32
imageH = 36
imageC = 1
labelDim = 1152

kW = 5
kH = 5
hStride = 1
vStride = 1
size1 = 25
wScale = 0.5
bValue = 0

scValue = 1
bnTimeConst = 0

features = ImageInput(imageW, imageH, imageC, imageLayout="$imageLayout$")
labels = Input((imageW:imageH:imageC), tag="label") 
featScale = Constant(0.0006)
featScaled = Scale(featScale, features)


# conv1
# weight[cMap1, kW1 * kH1 * inputChannels]
conv1 = ConvBNReLULayer(featScaled, cMap1, size1, kW, kH, hStride, vStride, wScale, bValue, scValue, bnTimeConst)

mse = SquareError(labels , conv1)
FeatureNodes = (features)
CriterionNodes = (mse)
The cntk file is as below

RootDir = ".."
ConfigDir = "$RootDir$/Config"
DataDir = "$RootDir$/Data"
OutputDir = "$RootDir$/Output"
ModelDir = "$OutputDir$/Models"

imageLayout = "cudnn"
deviceId = 0
command = train
precision = "float"
modelPath = "$ModelDir$/06_DeConvPooling"

stderr = "$OutputDir$/06_DeConvPooling_out"
traceLevel=1
numMBsToShowResult=1
prefetch=true
initOnCPUOnly=true

#######################################

TRAINING CONFIG

#######################################

train = [
action = "train"

BrainScriptNetworkBuilder = (new ComputationNetwork [ 
include "shared.bs"       
include "$ConfigDir$/06_DeConvPooling.bs"
])

SGD = [
    epochSize = 0
    minibatchSize = 17
    learningRatesPerMB = 0.0002
    momentumPerMB = 0.9
    maxEpochs = 50
]

reader = [
    readerType = "CNTKTextFormatReader"
    file = "$DataDir$/Train-Nodule-32x36.txt"
    input = [
        features = [
            dim = 1152
            format = "dense"
        ]
        labels = [
            dim = 1152
            format = "dense"
        ]
    ]
]   
]

Thank you!

**Comment by pinzhang10** [comment]
@frankseide 
Thank you for your quick reply!

the Shared.bs is downloaded from https://github.com/Microsoft/CNTK/blob/master/Examples/Image/MNIST/Config/Shared.bs

The 06_DeconvPooling.bs is as below (I am testing the new BrainScriptNetworkBuilder I made)

imageW = 32
imageH = 36
imageC = 1
labelDim = 1152

kW = 5
kH = 5
hStride = 1
vStride = 1
size1 = 25
wScale = 0.5
bValue = 0
scValue = 1
bnTimeConst = 0

features = ImageInput(imageW, imageH, imageC, imageLayout="$imageLayout$")
labels = Input((imageW:imageH:imageC), tag="label") 
featScale = Constant(0.0006)
featScaled = Scale(featScale, features)

conv1 = ConvBNReLULayer(featScaled, cMap1, size1, kW, kH, hStride, vStride, wScale, bValue, scValue, bnTimeConst)

mse = SquareError(labels , conv1)
FeatureNodes = (features)
CriterionNodes = (mse)
The cntk file is as below

RootDir = ".."
ConfigDir = "$RootDir$/Config"
DataDir = "$RootDir$/Data"
OutputDir = "$RootDir$/Output"
ModelDir = "$OutputDir$/Models"

imageLayout = "cudnn"
deviceId = 0
command = train
precision = "float"
modelPath = "$ModelDir$/06_DeConvPooling"

stderr = "$OutputDir$/06_DeConvPooling_out"
traceLevel=1
numMBsToShowResult=1
prefetch=true
initOnCPUOnly=true

#######################################

TRAINING CONFIG

#######################################

train = [
action = "train"

BrainScriptNetworkBuilder = (new ComputationNetwork [ 
include "shared.bs"

include "$ConfigDir$/06_DeConvPooling.bs"
])

SGD = [
epochSize = 0
minibatchSize = 17
learningRatesPerMB = 0.0002
momentumPerMB = 0.9
maxEpochs = 50
]

reader = [
readerType = "CNTKTextFormatReader"
file = "$DataDir$/Train-Nodule-32x36.txt"
input = [
features = [
dim = 1152
format = "dense"
]
labels = [
dim = 1152
format = "dense"
]
]
]

]

Thank you!

**Comment by pinzhang10** [comment]
@frankseide 
Thank you for your quick reply!

t**he Shared.bs is downloaded from** https://github.com/Microsoft/CNTK/blob/master/Examples/Image/MNIST/Config/Shared.bs

**The 06_DeconvPooling.bs is as below (I am testing the new BrainScriptNetworkBuilder I made)**

imageW = 32
imageH = 36
imageC = 1
labelDim = 1152

kW = 5
kH = 5
hStride = 1
vStride = 1
size1 = 25
wScale = 0.5
bValue = 0
scValue = 1
bnTimeConst = 0

features = ImageInput(imageW, imageH, imageC, imageLayout="$imageLayout$")
labels = Input((imageW:imageH:imageC), tag="label") 
featScale = Constant(0.0006)
featScaled = Scale(featScale, features)

conv1 = ConvBNReLULayer(featScaled, cMap1, size1, kW, kH, hStride, vStride, wScale, bValue, scValue, bnTimeConst)

mse = SquareError(labels , conv1)
FeatureNodes = (features)
CriterionNodes = (mse)


**The cntk file is as below**

RootDir = ".."
ConfigDir = "$RootDir$/Config"
DataDir = "$RootDir$/Data"
OutputDir = "$RootDir$/Output"
ModelDir = "$OutputDir$/Models"

imageLayout = "cudnn"
deviceId = 0
command = train
precision = "float"
modelPath = "$ModelDir$/06_DeConvPooling"

stderr = "$OutputDir$/06_DeConvPooling_out"
traceLevel=1
numMBsToShowResult=1
prefetch=true
initOnCPUOnly=true

#######################################

TRAINING CONFIG

#######################################

train = [
action = "train"

BrainScriptNetworkBuilder = (new ComputationNetwork [ 
include "shared.bs"

include "$ConfigDir$/06_DeConvPooling.bs"
])

SGD = [
epochSize = 0
minibatchSize = 17
learningRatesPerMB = 0.0002
momentumPerMB = 0.9
maxEpochs = 50
]

reader = [
readerType = "CNTKTextFormatReader"
file = "$DataDir$/Train-Nodule-32x36.txt"
input = [
features = [
dim = 1152
format = "dense"
]
labels = [
dim = 1152
format = "dense"
]
]
]

]

Thank you!

**Comment by pinzhang10** [comment]
I also tried Convolution() function and similar problem happened. Maybe this phenomenon can provide some clue. 

**## 1. If I run ConvBNReLULayer(), the error is:**
`##############################################################################
#                                                                            #
# Action "train"                                                             #
#                                                                            #
##############################################################################

CNTKCommandTrainBegin: train

Creating virgin network.

[CALL STACK]
      conv1 = ConvBNReLULayer(features, cMap1, size1, kW, kH, hStride, vStride, wScale, bValue, scValue, bnTimeConst)
                             a
  ].y
    9
      y = RectifiedLinear(c)
                         8
      Relu(_, tag='') = new ComputationNode [ operation = 'RectifiedLinear' ; inputs = _ /*plus the function args*/ ]
                        7                                                              6
      y = RectifiedLinear(c)
                          5
      c = ConvBNLayer(inp, outMap, inWCount, kW, kH, hStride, vStride, wScale, bValue, scValue, bnTimeConst)
                     4
  ].c
    3
      c = ConvBNLayerW(W, inp, outMap, kW, kH, hStride, vStride, bValue, scValue, bnTimeConst)
                      2
  ].y
    1
      y = BatchNormalization(c, sc, b, m, isd, eval=false, spatial=true, normalizationTimeConstant=bnTimeConst /* , imageLayout=$imageLayout$*/)
                            0^
EXCEPTION occurred while evaluating: Shared.bs(67): function expects 6 positional parameters, 5 were provided
`

**## If I run Convolution(), the error is** 
`##############################################################################
#                                                                            #
# Action "train"                                                             #
#                                                                            #
##############################################################################

CNTKCommandTrainBegin: train

Creating virgin network.

[CALL STACK]
  network = (new ComputationNetwork [ 
             1
      out = Convolution(w, features, (kW:kH:imageC), cMap1, stride=(hStride:vStride:imageC), sharing=(true:true:true), autoPadding=false, lowerPad=0, upperPad=0, transpose=0, imageLayout="CHW")
                       0^
EXCEPTION occurred while evaluating: ../Config/06_DeConvPooling.bs(23): function expects 3 positional parameters, 4 were provided`


**## The 06_DeConvPooling.cntk code is as below:** 

`RootDir = ".."
ConfigDir = "$RootDir$/Config"
DataDir   = "$RootDir$/Data"
OutputDir = "$RootDir$/Output"
ModelDir  = "$OutputDir$/Models"
deviceId = 0
command = train
precision = "float"
modelPath = "$ModelDir$/06_DeConvPooling"
stderr = "$OutputDir$/06_DeConvPooling_out"
traceLevel=1
numMBsToShowResult=1
prefetch=true
initOnCPUOnly=true

#######################################
#  TRAINING CONFIG                    #
#######################################

train = [
    action = "train"

    BrainScriptNetworkBuilder = (new ComputationNetwork [ 
	include "Shared.bs"       
    include "$ConfigDir$/06_DeConvPooling.bs"
])
    
    SGD = [
        epochSize = 0
        minibatchSize = 17
        learningRatesPerMB = 0.0002
        momentumPerMB = 0.9
        maxEpochs = 50
    ]
    
    reader = [
        readerType = "CNTKTextFormatReader"
        file = "$DataDir$/Train-Nodule-32x36.txt"
        input = [
            features = [
                dim = 1152
                format = "dense"
            ]
            labels = [
                dim = 1152
                format = "dense"
            ]
        ]
    ]   
]`

**## The 06_DeConvPooling.bs code is as below:** 
`
    imageW = 32
    imageH = 36
    imageC = 1
    labelDim = 1152
 
    kW = 5
    kH = 5
    hStride = 1
    vStride = 1
    size1 = 25

    wScale = 0.5
    bValue = 0
	scValue = 1
    bnTimeConst = 0

    features = ImageInput(imageW, imageH, imageC, imageLayout="CHW")

    conv1 = ConvBNReLULayer(features, cMap1, size1, kW, kH, hStride, vStride, wScale, bValue, scValue, bnTimeConst)

	w = ConvW(cMap1, size1, wScale)
    out = Convolution(w, features, (kW:kH:imageC), cMap1, stride=(hStride:vStride:imageC), sharing=(true:true:true), autoPadding=false, lowerPad=0, upperPad=0, transpose=0, imageLayout="CHW")

    mse = SquareError(features, features)
    FeatureNodes = (features)
    CriterionNodes = (mse)`


I tried all the methods I can come up with to fix it. But the problem is still there. Really appreciate your help!

**Comment by pinzhang10** [comment]
@frankseide 

Problem solved!

The reason is: 
Using NDLNetworkBuilder, when we call the functions like Convolution() and BatchNormalization(), we are using the functions in dll or lib. But using BrainScriptNetworkBuilder, when we call those functions, we are actually using the definitions in CNTK.core.bs. The declarations of these functions are different in source code and CNTK.core.bs. That is why I got errors.

Therefore, after updating from NDLNetworkBuilder to BrainScriptNetworkBuilder, if some function does not work, it is probably cased by the different declaration of the function with the same name in CNTK.core.bs.

I changed the ConvBNReLULayer into the definition below and it works fine.

`ConvBNReLULayer(inp, outMap, kW, kH, inMap, hStride, vStride, wScale, bValue, scValue, bnTimeConst) = [
    inWCount = kW * kH * inMap
	w = LearnableParameter (outMap, inWCount, learningRateMultiplier = 1.0, init = 'gaussian'/*|fixedValue|gaussian|fromFile|fromLiteral*/, initValueScale = wScale, value = 0, initFromFilePath = '', initFromLiteral = '', initOnCPUOnly=true, randomSeed=-1, tag='')
    c = Convolution(w, inp, (kW:kH:inMap), mapDims=outMap, stride=(hStride:vStride:inMap), sharing = true, autoPadding = true, lowerPad = 0, upperPad = 0, transpose=false, imageLayout='CHW', maxTempMemSizeInSamples = 0, tag='')
	
	b   = Parameter(outMap, 1, init="fixedValue", value=bValue)
    sc  = Parameter(outMap, 1, init="fixedValue", value=scValue)
    m   = Parameter(outMap, 1, init="fixedValue", value=0, learningRateMultiplier=0)
    isd = Parameter(outMap, 1, init="fixedValue", value=0, learningRateMultiplier=0)
	y = BatchNormalization(c, sc, b, m, isd, true, normalizationTimeConstant = 0, blendTimeConstant = 0, epsilon = 0.00001, useCntkEngine = true, imageLayout='CHW', tag='')
    
	out = RectifiedLinear(y)
].out`

**Comment by pinzhang10** [comment]
@frankseide

Problem solved!

The reason is: 
Using NDLNetworkBuilder, when we call the functions like Convolution() and BatchNormalization(), we are using the functions in dll or lib. But using BrainScriptNetworkBuilder, when we call those functions, we are actually using the definitions in CNTK.core.bs. The declarations of these functions are different in source code and CNTK.core.bs. That is why I got errors.

Therefore, after updating from NDLNetworkBuilder to BrainScriptNetworkBuilder, if some function does not work, it is probably cased by the different declaration of the function with the same name in CNTK.core.bs.

I changed the ConvBNReLULayer into the definition below and it works fine.

ConvBNReLULayer(inp, outMap, kW, kH, inMap, hStride, vStride, wScale, bValue, scValue, bnTimeConst) = [
inWCount = kW * kH * inMap
w = LearnableParameter (outMap, inWCount, learningRateMultiplier = 1.0, init = 'gaussian'/|fixedValue|gaussian|fromFile|fromLiteral/, initValueScale = wScale, value = 0, initFromFilePath = '', initFromLiteral = '', initOnCPUOnly=true, randomSeed=-1, tag='')
c = Convolution(w, inp, (kW:kH:inMap), mapDims=outMap, stride=(hStride:vStride:inMap), sharing = true, autoPadding = true, lowerPad = 0, upperPad = 0, transpose=false, imageLayout='CHW', maxTempMemSizeInSamples = 0, tag='')
b   = Parameter(outMap, 1, init="fixedValue", value=bValue)
sc  = Parameter(outMap, 1, init="fixedValue", value=scValue)
m   = Parameter(outMap, 1, init="fixedValue", value=0, learningRateMultiplier=0)
isd = Parameter(outMap, 1, init="fixedValue", value=0, learningRateMultiplier=0)
y = BatchNormalization(c, sc, b, m, isd, true, normalizationTimeConstant = 0, blendTimeConstant = 0, epsilon = 0.00001, useCntkEngine = true, imageLayout='CHW', tag='')
out = RectifiedLinear(y)
].out
